Src/FluentAssertions/Primitives/StringEqualityStrategy.cs

```diff
@@ -57,19 +58,47 @@ private string GetMismatchSegmentForStringsOfDifferentLengths(string subject, st
 
     public void ValidateAgainstMismatch(IAssertionScope assertion, string subject, string expected)
     {
-        if (!ValidateAgainstSuperfluousWhitespace(assertion, subject, expected) ||
-            !ValidateAgainstLengthDifferences(assertion, subject, expected))
+        if (!ValidateAgainstSuperfluousWhitespace(assertion, subject, expected))
         {
             return;
         }
 
-        int indexOfMismatch = subject.IndexOfFirstMismatch(expected, comparisonMode);
-
-        if (indexOfMismatch != -1)
+        if (expected.IsLongOrMultiline() || subject.IsLongOrMultiline())
         {
+            int indexOfMismatch = subject.IndexOfFirstMismatch(expected, comparisonMode);
+
+            if (indexOfMismatch == -1)
+            {
+                ValidateAgainstLengthDifferences(assertion, subject, expected);
+                return;
+            }
+
+            string lineInfo = $"at index {indexOfMismatch}";
```

Comment:
🤔 I think naming this `locationDescription` or something like that will help explain the purpose

Reply:
Renamed to `locationDescription` in [9becb00](https://github.com/fluentassertions/fluentassertions/pull/2307/commits/9becb00b59cf8e3fcb7ed2bfdec5b05582b2b44b).